src/object/obj_enum.c

```diff
@@ -559,11 +634,8 @@ fill_rec(daos_handle_t ih, vos_iter_entry_t *key_ent, struct dss_enum_arg *arg,
 			iod_size = key_ent->ie_gsize;
 			if (iod_size == key_ent->ie_rsize)
 				data_size = iod_size;
-			else
-				data_size = 0;
 		} else {
 			iod_size = key_ent->ie_rsize;
-			data_size = iod_size * key_ent->ie_recx.rx_nr;
```

Comment:
you mean for ITER_RECX, always don't inline it even if it is very small?

Reply:
yes, if we pack inline data, which might cause extra complexity for epoch merging. Since more the small size data comes from single value, so let's disable the inline data packing for array, i.e. extra RPC for small RECX data.